validations/transfer.test.ts

```diff
@@ -0,0 +1,136 @@
+import { expect, use } from 'chai'
+import { Wallet, ethers } from 'ethers'
+import path from 'path'
+import { randomUUID } from 'crypto'
+import { AddKycParams, FiatConnectClient } from '@fiatconnect/fiatconnect-sdk'
+import {
+  KycSchema,
+  TransferStatus,
+  Network,
+} from '@fiatconnect/fiatconnect-types'
+import { config } from '../src/config'
+import { chaiPlugin } from 'api-contract-validator'
+import { MOCK_QUOTE } from '../src/mock-data/quote'
+import { MOCK_KYC } from '../src/mock-data/kyc'
+import { MOCK_FIAT_ACCOUNTS } from '../src/mock-data/fiat-account'
+
+const apiDefinitionsPath = path.join(config.openapiSpec)
+use(chaiPlugin({ apiDefinitionsPath }))
+
+describe('/transfer', () => {
+  const mockAccountData =
+    MOCK_FIAT_ACCOUNTS[
+    config.fiatAccountMock as keyof typeof MOCK_FIAT_ACCOUNTS
+  ]
+
+  const mockKYCInfo: AddKycParams<KycSchema> =
+    MOCK_KYC[config.kycMock as keyof typeof MOCK_KYC]
+
+  describe('/in', () => {
+    const wallet = Wallet.createRandom()
+
+    const fiatConnectClient = new FiatConnectClient(
+      {
+        baseUrl: config.baseUrl,
+        network: Network.Alfajores,
+        accountAddress: wallet.address,
+      },
+      (message: string) => wallet.signMessage(message),
+    )
+
+    const quoteInParams = {
+      ...MOCK_QUOTE[config.quoteInMock],
+      address: wallet.address,
+    }
+
+    it('able to transfer fiat in for crypto', async () => {
+      const loginResult = await fiatConnectClient.login()
+      expect(loginResult.isOk).to.be.ok
+
+      const quoteInResponse = await fiatConnectClient.createQuoteIn(
+        quoteInParams,
+      )
+      expect(quoteInResponse.isOk).to.be.ok
```

Comment:
the spec does not require providers to support transfers in, so this is not a fair test. To my knowledge, no provider right now actually supports transfers of both types.

Reply:
Instead, we should probably infer from the configuration that if a quote in mock is provided, then transfers in are supported, and if it is not, then they are not and we should expect the appropriate error type from the spec (I think there is an error specifically for transfer type not supported..)

Reply:
There currently is not a general "transfer type not supported" error in the spec, although it's a good idea - it's more honest than returning an e.g. `GeoNotSupported` error. Happy to add this in to the spec, though in the meantime, it might be better to check that it returns _any_ error? Or just skip tests for quote in if a mock isn't provided?

Reply:
Yeah for now I'll skip the transfer in test if they haven't specified a quote in mock.